src/metavision_wrapper.cpp

```diff
@@ -428,8 +442,44 @@ void MetavisionWrapper::setDecodingEvents(bool decodeEvents)
   }
 }
 
+void MetavisionWrapper::activateTrailFilter()
+{
+  Metavision::I_EventTrailFilterModule * i_trail_filter =
+    cam_.get_device().get_facility<Metavision::I_EventTrailFilterModule>();
+
+  Metavision::I_EventTrailFilterModule::Type filter_type;
+
+  /*
+  if (trailFilter_.type == TrailFilter::Type::TRAIL) {
+    filter_type = Metavision::I_EventTrailFilterModule::Type::TRAIL;
+  } else if (trailFilter_.type == TrailFilter::Type::STC_CUT_TRAIL) {
+    filter_type = Metavision::I_EventTrailFilterModule::Type::STC_CUT_TRAIL;
+  } else if (trailFilter_.type == TrailFilter::Type::STC_KEEP_TRAIL) {
+    filter_type = Metavision::I_EventTrailFilterModule::Type::STC_KEEP_TRAIL;
+  }
+  */
+  const auto it = trailFilterMap.find(trailFilter_.type);
+  if (it == trailFilterMap.end()) {
+    LOG_WARN_NAMED("unknown trail filter type " << trailFilter_.type);
```

Comment:
call BOMB_OUT() here (logs and throws exception). As it stands, this is a bug because if you don't abort here, you will be dereferencing an invalid iterator.